python/ray/train/torch/config.py

```diff
@@ -207,4 +214,13 @@ def on_shutdown(self, worker_group: WorkerGroup, backend_config: TorchConfig):
     def on_training_start(
         self, worker_group: WorkerGroup, backend_config: BackendConfig
     ):
+        device_manager = get_torch_device_manager_cls_by_resources(
+            self.resources_per_worker
```

Comment:
Let's just check if `torch.cuda.is_available()`, etc. in this instead of searching through the resources.

Reply:
That way, there's no need to pass in the `resources_per_worker`

Reply:
We think it's more suitable to select a `device_manager` from explicit accelerator configurations. However, to minimize modifications to modules unrelated to the framework, such as the `backend`, we have changed to use `ray.get_runtime_context().get_accelerator_id` to get the device tag, reducing the need for the parameter passing. After that, the `device_manager` will use APIs like `torch.cuda.is_available()` to check the availability of devices.